apps/linode-marketplace-jupyterlab/roles/post/tasks/main.yml

```diff
@@ -0,0 +1,21 @@
+# motd and creds gen
+- name: copy MOTD template to /etc/motd
+  template:
+    src: templates/motd.j2
+    dest: /etc/motd
+
+- name: writing docker creds into file
+  copy:
+    dest: /root/.credentials
+    mode: '0600'
+    content: |
+      Sudo Username: {{ username }}
+      Sudo Password: {{ password }}
+
+# Retrieving and storing Jupyter token
+- name: Retrieve and store Jupyter token
+  command: "{{ item }}"
+  loop:
+    - "docker logs jupyter_jupyterlab_1 2>&1 | grep -oP '(?<=token=)[^ ]+' | head -n 1 >> /root/.credentials"
+    - "chmod 0600 /root/.credentials"
+  ignore_errors: true
```

Comment:
It is generally best practice to always use Ansible modules / builtins if you can vs. running a straight shell command through it. 
I would recommend rewriting this command loop into a couple of tasks so you can use Ansible. Something like: 
```
- name: std out docker logs consul 
  ansible.builtin.shell: docker logs jupyter_jupyterlab_1
  register: result
 # until handles docker catching up with ansible. something like this is preferable to ignore_errors true
  until: "result is not failed"
  retries: 3
  delay: 3

- name: set jupyter token fact 
  ansible.builtin.set_fact: 
    jupyter_token: "{{ result.stdout | regex_search('(?<=token=)[^ ]+') }}"
    cacheable: yes

- name: writing creds into file
  copy:
    dest: /root/.credentials
    mode: '0600'
    content: |
      Sudo Username: {{ username }}
      Sudo Password: {{ password }}
      Jupyter Token: {{ jupyter_token }}
```